codersdk/buildinfo.go

```diff
@@ -16,6 +17,18 @@ type BuildInfoResponse struct {
 	Version string `json:"version"`
 }
 
+// TrimmedVersion trims build information from the version.
+// E.g. 'v0.7.4-devel+11573034' -> 'v0.7.4'.
+func (b BuildInfoResponse) TrimmedVersion() string {
+	// Linter doesn't like strings.Index...
+	idx := bytes.Index([]byte(b.Version), []byte("-devel"))
+	if idx < 0 {
+		return b.Version
+	}
+
+	return b.Version[:idx]
+}
```

Comment:
We can do this with `semver` too

Reply:
Unfortunately `semver.Canonical` includes the pre-release so `v0.7.4-devel+11573034` becomes `v0.7.4-devel`. Which is reasonable but doesn't work well for what I'm trying to achieve here.

Reply:
How about we just rewrite `v0.7.4-devel+11573034` => `v0.7.4+devel.11573034` via simple string replacement before passing it along to semver? Ideally we would change our version output in the build stage though.

Reply:
Fair enough it's honestly too much discussion on my part for such a small change. It's been implemented